### PR TITLE
add `allow_sending_without_reply` param to methods

### PR DESCRIPTION
**Pull request by MyMindWorld**
Issue #167

Besides i removed some unused params i found along the way and refactored hardcoded strings to ApiConstants

### CODE REVIEW

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SendMediaGroupIT.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ class SendMediaGroupIT : ApiClientIT() {
         val expectedRequestBody = String.format(
             getFileAsStringFromResources<SendMediaGroupIT>("sendMediaGroupRequestBody1.txt"),
             multipartBoundary
-        )
+        ).trimIndent()
```
Comment (vjgarciag96) [inline telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SendMediaGroupIT.kt:44]: Why is this change needed?

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SetWebhookIT.kt [hunk 2]
```diff
@@ -96,7 +96,7 @@ class SetWebhookIT : ApiClientIT() {
         val expectedRequestBody = String.format(
             getFileAsStringFromResources<SetWebhookIT>("setWebhookMultipartWithCertificateRequestBody.txt"),
             multipartBoundary
-        )
+        ).trimIndent()
```
Comment (vjgarciag96) [inline telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SetWebhookIT.kt:99]: Same as before - why is this change needed?

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/types/TelegramBotResultTest.kt [hunk 3]
```diff
@@ -61,7 +61,7 @@ class TelegramBotResultTest {
     }
```
Comment (vjgarciag96) [inline telegram/src/test/kotlin/com/github/kotlintelegrambot/types/TelegramBotResultTest.kt:61]: I would keep the previous test names. One test name or other is mostly a matter of taste, and is a change unrelated to the PR, so I would advocate to keep it as was. Otherwise, we could end up with back and forth changes in tests names that lead nowhere.

**Review by vjgarciag96 — COMMENTED** [review]
Thank you so much for the contribution @MyMindWorld, good job!

I added a couple of minor comments in the PR.

I can see that there are two additional API methods that also have this new parameter, `sendGame` and `sendMediaGroup`. We don't have the `sendGame` method in the library yet, so I have created a new issue to keep that in the radar (https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/issues/187). And the `sendMediaGroup` is a bit special compared to the other methods, so I've also created an issue to add the parameter there as a separated task (https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/issues/188).

Regarding the PR, I like the cleanup changes about using the API constants. It would be nice if we could keep the cleanup commits separated from the others for clarity, to keep a better git history and to make reviewing easier. I mean, it's fine and we can go ahead with them as they're here, but I'm just commenting this in case it can be helpful for future PRs.

Thanks again for the contribution! 🙂

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SendMediaGroupIT.kt (hunk 1 above)
Comment (MyMindWorld) [reply telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SendMediaGroupIT.kt:44]: On windows machine we receive content with indent in the beginning, In real-life application they won't matter. 
image
If there is more to dig, please point where to)

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/types/TelegramBotResultTest.kt (hunk 3 above)
Comment (MyMindWorld) [reply telegram/src/test/kotlin/com/github/kotlintelegrambot/types/TelegramBotResultTest.kt:61]: What do you think than of separate pr with naming convention?
I belive that tests naming should follow one 
e.g. `Should do something when given` or `When given should do something`
This way we can detect some issues with architecrure or split tests with similar setups into nested classes
e.g.
```kotlin
@Nested
inner class `Should send MediaGroup with media group composed by a video(fileUrl), a photo(fileId)` {
    // shared fixture with response and media group
    @Test
    fun `when all mandatory arhuments is passed`() {
        // test
    }
    @Test
    fun `when all arguments is passed`() {
        // test
    }
```

Comment (vjgarciag96) [reply telegram/src/test/kotlin/com/github/kotlintelegrambot/types/TelegramBotResultTest.kt]: I don't have a strong opinion about tests naming. I used to follow the "should do something when given" naming but it sometimes ends up being too verbose and makes test names and tests harder to understand. Now I try to keep it simple in the name and just follow a `given-when-then`|`arrange-act-assert` structure in the test itself. I guess it's also a matter of taste and something to be agreed with all the people working in a project to be used in a consistent way (in this case the project is quite small and the people working on it is not too many, so I guess it's not that critical).

I like the nested tests though, we're not taking advantage of them and we should, good call 👍🏻

**Review by vjgarciag96 — APPROVED** [review]
Let's merge this!

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SendMediaGroupIT.kt (hunk 1 above)
Comment (vjgarciag96) [reply telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SendMediaGroupIT.kt:44]: Oh gotcha, good to know!

File: telegram/src/test/kotlin/com/github/kotlintelegrambot/types/TelegramBotResultTest.kt (hunk 3 above)
Comment (MyMindWorld) [reply telegram/src/test/kotlin/com/github/kotlintelegrambot/types/TelegramBotResultTest.kt]: Maybe than we should create issue with poll about it? Also it would help with writing contributing guide for project, it could decrease number of this **taste** cases
Thanks for review!)
